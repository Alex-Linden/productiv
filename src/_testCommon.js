const TODO_CLASS = ".Todo";

const TEST_TODOS = [{
  id: 11,
  title: "test1",
  description: "test1",
  priority: 1,
},
{
  id: 22,
  title: "test2",
  description: "test2",
  priority: 2,
},
{
  id: 33,
  title: "test3",
  description: "test3",
  priority: 3,
}];

export { TEST_TODOS, TODO_CLASS };