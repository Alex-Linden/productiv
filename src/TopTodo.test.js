import React from "react";
import{TEST_TODOS} from "./_testCommon";
import TopTodo from "./TopTodo";
import { render } from "@testing-library/react";

describe("TopTodo component", function () {
  it("renders without crashing", function () {
    render(<TopTodo todos={TEST_TODOS} />);
  });

  it("matches snapshot", function () {
    const { container } = render(<TopTodo todos={TEST_TODOS} />);
    expect(container).toMatchSnapshot();
  });

  it("displays Top todo data", function () {
    const { container } = render(<TopTodo todos={TEST_TODOS} />);

    expect(container.querySelector(".Todo")).toBeInTheDocument();
    expect(container.querySelector(".Todo"))
      .toContainHTML("<b>test1</b>");
  });
});