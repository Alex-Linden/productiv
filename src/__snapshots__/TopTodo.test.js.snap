// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTodo component matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        test1
      </b>
       
      <small>
        (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        test1
      </small>
    </div>
  </div>
</div>
`;
